### PR TITLE
Board games update: chess edition

### PR DESCRIPTION
**Pull request by Alberyk**
-updates the board to be a little better, porting bay updates to it
-adds chess pieces as well, ported from bay too
-you can now drag the board to yourself to hold it, instead of having to drag it everywhere if you want to move it
-added chess and checkers kits to the loadout

### CODE REVIEW

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/obj/pieces.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
pawn_black||pawn_black|Created
bishop_black||bishop_black|Created
king_black||king_black|Created
queen_black||queen_black|Created
knight_black||knight_black|Created
rook_black||rook_black|Created
pawn_red||pawn_red|Created
bishop_red||bishop_red|Created
king_red||king_red|Created
queen_red||queen_red|Created
knight_red||knight_red|Created
rook_red||rook_red|Created
</details>

**Comment by chaoko99** [comment]
The bay chessboard doesn't work by the way. I just feel like mentioning that.

File: code/modules/games/boardgame.dm [hunk 1]
```diff
@@ -24,6 +23,8 @@
 	..()
 
 /obj/item/weapon/board/attack_hand(mob/living/carbon/human/M as mob)
+	if(!istype(loc, /turf/)) //so if you want to play the game, you need to put it down somewhere
```
Comment (skull132) [inline code/modules/games/boardgame.dm]: `if (!isturf(loc))`

File: code/modules/games/boardgame.dm [hunk 2]
```diff
@@ -9,12 +9,11 @@
...
 	var/selected = -1
 
-/obj/item/weapon/board/New()
-	..()
-	var i
-	for(i = 0; i < 12; i++)
-		new /obj/item/weapon/checker(src.loc)
-		new /obj/item/weapon/checker/red(src.loc)
+/obj/item/weapon/board/MouseDrop(mob/user as mob)
+	if((user == usr && (!use_check(usr))) && (usr.contents.Find(src) || in_range(src, usr)))
```
Comment (BurgerLUA) [inline code/modules/games/boardgame.dm]: Are we still using usr? I thought that was no longer a thing anymore.

**Pull request merged**